commit b75a06993e820005463d87fec47e7a1b44f9c674 (refs/changes/96/3596/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-03-12T15:43:07-07:00 (1 year, 7 months ago)
    
    TRITON-232 vmapi should not set refreservation for bhyve VMs
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/common/validation.js b/lib/common/validation.js
index 2d0f322..bf86045 100644
--- a/lib/common/validation.js
+++ b/lib/common/validation.js
@@ -2478,13 +2478,17 @@ exports.setDefaultValues = function (params, options) {
         params.disks[0].image_name = params.image.name;
         params.disks[0].image_size = params.image.image_size;
 
-        // Set a default refreservation for i > 0 disks
-        for (i = 1; i < params.disks.length; i++) {
-            if (params.disks[i].refreservation === undefined) {
-                if (config.reserveKvmStorage === false) {
-                    params.disks[i].refreservation = 0;
-                } else {
-                    params.disks[i].refreservation = params.disks[i].size;
+        // We don't set the refreservation via VMAPI for bhyve and instead
+        // allow SmartOS to determine the refreservation.
+        if (params.brand === 'kvm') {
+            // Set a default refreservation for i > 0 disks
+            for (i = 1; i < params.disks.length; i++) {
+                if (params.disks[i].refreservation === undefined) {
+                    if (config.reserveKvmStorage === false) {
+                        params.disks[i].refreservation = 0;
+                    } else {
+                        params.disks[i].refreservation = params.disks[i].size;
+                    }
                 }
             }
         }
diff --git a/package.json b/package.json
index b4a2619..2f7d67b 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.4.0",
+  "version": "9.5.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
